### PR TITLE
Fix issue #19: emtpy new git file

### LINKED ISSUE #19: git diff adding a new empty file is not parsed correctly
a git diff adding a new empty file is not parsed correctly. the patch set contains 0 items in that case, while it should contain one.

sample:

``` diff
diff --git a/empty b/empty
new file mode 100644
index 0000000..e69de29
```

### PR DESCRIPTION
**Pull request by Felixoid**
Hello. I've faced the issue today in ClickHouse/ClickHouse#32911 that the empty new files aren't fetched. It means, our CI may not work correctly for some other edge cases.

Here I've implemented the parsing for such diffs and tests for them:

```diff
diff --git a/boo.bin b/boo.bin
new file mode 100644
index 0000000..ae000000
```

The PR fixes #19

### CODE REVIEW

**Comment by Felixoid** [comment]
Hi @matiasb,
Happy New Year!

Do you mind taking a look here? It's a kinda annoying issue for us. We use your library in a CI, and it's blind for empty new files currently.

**Comment by matiasb** [comment]
Hi @Felixoid, thanks, happy new year for you too!
I will take a look during the week, still catching up with multiple things after being a few weeks off.

Thanks for the PR :+1:

File: unidiff/patch.py [hunk 1]
```diff
@@ -473,8 +475,10 @@ def _parse(self, diff, encoding, metadata_only):
             # check for a git file rename
             is_diff_git_header = RE_DIFF_GIT_HEADER.match(line)
             if is_diff_git_header:
-                if patch_info is None:
-                    patch_info = PatchInfo()
+                if current_file is not None and not is_file_appended:
```
Comment (matiasb) [inline unidiff/patch.py:478]: I think we could refactor this to avoid introducing the `is_file_appended` flag by always setting up a `PatchedFile` when a git header is found (and some minor changes to accommodate for that), will check.

**Review by matiasb — APPROVED** [review]
Looking good, I will try some refactoring after merging (added inline notes, fyi).
Thanks!

File: unidiff/patch.py [hunk 2]
```diff
@@ -63,7 +64,7 @@ def implements_to_string(cls):
         return cls
 else:
     from io import StringIO
-    from typing import Iterable, Optional, Union
+    from typing import Iterable, Optional, Union, Tuple
```
[the remark this answers is not in the document]
Comment (Felixoid) [reply unidiff/patch.py:67]: Right, it was in the middle of playing around, I guess

File: unidiff/patch.py (hunk 1 above)
Comment (Felixoid) [reply unidiff/patch.py:478]: AFAI remember I was in a rush of fixing it. Most probably, there is a more elegant way to fix it. I am 100% sure you are more confident with code! =)

**Comment by matiasb** [comment]
Merging, thanks :+1: 
I have some refactoring in progress, will land afterwards and prepare a release.
